src/StepAdaptationController/src/StepAdaptationController.cpp

```diff
@@ -487,9 +517,364 @@ bool StepAdaptationController::getAdaptatedFootTrajectory(const footTrajectoryGe
+                output.pushRecoveryActiveIndex++;
+                yInfo()<<"triggering the push recovery";
+                if((abs(input.dcmPositionSmoothed(0) - getEstimatedDCM()(0))) > getDCMErrorThreshold()(0))
+                {
+                    tempDCMError(0)=getEstimatedDCM()(0);
+                }
+                if((abs(input.dcmPositionSmoothed(1) - getEstimatedDCM()(1))) > getDCMErrorThreshold()(1))
+                {
+                    tempDCMError(1)=getEstimatedDCM()(1);
+                }
+                setCurrentDcmPosition(tempDCMError);
+            }
+            else
+            {
+                output.pushRecoveryActiveIndex++;
+                setCurrentDcmPosition(output.dcmPositionAdjusted.front());
+            }
+
+        }
+        else
+        {
+            if (output.pushRecoveryActiveIndex<=m_pushRecoveryActivationIndex)
+            {
+                output.pushRecoveryActiveIndex=0;
+                setCurrentDcmPosition(output.dcmPositionAdjusted.front());
+            }
+            else if(output.pushRecoveryActiveIndex==(m_pushRecoveryActivationIndex+1))
+            {
+                setCurrentDcmPosition(output.dcmPositionAdjusted.front());
+                output.pushRecoveryActiveIndex++;
+            }
+            else
+            {
+                setCurrentDcmPosition(output.dcmPositionAdjusted.front());
+            }
+        }
+
+        iDynTree::Vector2 dcmAtTimeAlpha;
+        double timeAlpha = (secondDS->getTrajectoryDomain().second + secondDS->getTrajectoryDomain().first) / 2;
+
+        if(!input.dcmSubTrajectories[numberOfSubTrajectories-2]->getDCMPosition(timeAlpha, dcmAtTimeAlpha, checkFeasibility))
+        {
+            yError() << "[StepAdaptationController::runStepAdaptation] unable to get DCM Position ";
+            return false;
+        }
+
+        iDynTree::Vector2 nominalDcmOffset;
+        iDynTree::toEigen(nominalDcmOffset) = iDynTree::toEigen(dcmAtTimeAlpha) - iDynTree::toEigen(nextZmpPosition);
+        setNominalDcmOffset(nominalDcmOffset);
+
+        //timeOffset is the time of start of this step(that will be updated in updateTrajectory function at starting point of each step )
+        setTimings(input.omega, input.time - input.timeOffset, firstSS->getTrajectoryDomain().second,
+                                    secondDS->getTrajectoryDomain().second - secondDS->getTrajectoryDomain().first);
+
+        SwingFoot swingFoot;
+        if (!input.leftInContact.front())
+        {
+        swingFoot=SwingFoot::Left;
+        }
+        else
+        {
+        swingFoot=SwingFoot::Right;
+        }
+
+        if(!solve(swingFoot))
+        {
+            yError() << "[StepAdaptationController::runStepAdaptation] unable to solve the step adjustment optimization problem";
+            return false;
+        }
+
+        output.impactTimeNominal = firstSS->getTrajectoryDomain().second + input.timeOffset;
+        if(output.pushRecoveryActiveIndex==(5+1))
```

Comment:
Could you explain to me what ``pushRecoveryActiveIndex`` and why 5+1?

Reply:
The `5` was there by mistake! now I replace it by "m_pushRecoveryActivationIndex"! "m_pushRecoveryActivationIndex" is an integer threshold that shows how many sequential control cycles of push detection should happen to active the step adaptation controller, so when we are only one control cycle after the 5 times of detecting the push, the step adaptation will be activated!